pkg/flow/adapter/xslttransform/adapter_test.go

```diff
@@ -256,6 +264,60 @@ func TestXSLTTransformEvents(t *testing.T) {
 	}
 }
 
+func TestXSLTTransformKSINK(t *testing.T) {
+	testCases := map[string]struct {
+		allowXSLTOverride bool
+		xslt              string
+
+		inEvent                cloudevents.Event
+		expectedStringResponse string
+		expectPanic            string
+		expectCategory         string
+	}{
+		"transform ok": {
+			allowXSLTOverride:      false,
+			xslt:                   tXSLT,
+			inEvent:                newCloudEvent(tXML, cloudevents.ApplicationXML),
+			expectedStringResponse: tExpectedResponseString,
+			expectCategory:         tSuccessAttribute,
+		},
+	}
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			svr := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				body, err := ioutil.ReadAll(r.Body)
+				assert.NoError(t, err)
+				assert.Equal(t, tc.expectedStringResponse, string(body))
```

Comment:
This looks clumsy to me.
The server should simply save the requests somewhere without handling the data (some slice that we append to), and assertions should be performed on these at the end of the test.

Even simpler: Knative has a mock CE client that records the events it "sends", and allows you to inspect these events. You can see it in action in tests for source adapters.

Reply:
I saw the mock CE Client in the sources.. but we have to set a K_SINK URL in order for them to send to the K_SINK.

Reply:
also.. the assertions are at the top.. however they are performed technically at the end, because that call is not made until the end of the test.

Reply:
But this test _is_ for the K_SINK scenario, right?
The mock client doesn't technically use the sink, because it's mocked.
I can't see why we couldn't use it here instead of starting a HTTP server.

Reply:
I got it figured out :) seems to be working okay.